File: src/Stack.py
# Zifan Yang
# zyang45@jhu.edu

class Stack(object):
	elements = []

	def __init__(self):
		self.elements = []

	def isEmpty(self):
		return len(self.elements) == 0

	def peek(self):
		return self.elements[self.size()-1]

	def size(self):
		return len(self.elements)

	def push(self, obj):
		# print("PUSH: "+str(obj))
		self.elements.append(obj)

	def pop(self):
		# print("POP: "+str(self.elements[self.size()-1]))
		return self.elements.pop()


# print("test stack")
# sk = Stack()
# sk.push(11)
# mylist = []
# mylist.append('a')
# mylist.append('b')
# sk.push(mylist)
# print(sk.pop())
# print(sk.pop())

File: SimpleCompiler/SimpleCompiler/Stack.swift
//
//  Stack.swift
//  SimpleCompiler
//
//  Created by Zifan  Yang on 4/12/18.
//  Copyright © 2018 Zifan  Yang. All rights reserved.
//

import Foundation
class Stack {
    var stack: [AnyObject]
    init() {
        stack = [AnyObject]()
    }
    func push(object: AnyObject) {
        stack.append(object)
    }
    func pop() -> AnyObject? {
        if !isEmpty() {
            return stack.removeLast()
        } else {
            return nil
        }
    }
    func isEmpty() -> Bool {
        return stack.isEmpty
    }
    func peek() -> AnyObject? {
        return stack.last
    }
    func size() -> Int {
        return stack.count
    }
}
